### PR TITLE
Support pidfd_open syscall

### PR DESCRIPTION
**Pull request by Rohith-Raju**
**What type of PR is this?**

/kind feature

**Any specific area of the project related to this PR?**

/area driver-kmod

/area driver-bpf

/area driver-modern-bpf

/area tests

**Does this PR require a change in the driver versions?**

**What this PR does / why we need it**:

This PR adds support for `pidfd_open` syscall 

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Support pidfd_open syscall
```

### CODE REVIEW

File: test/drivers/test_suites/syscall_exit_suite/pidfd_open_x.cpp [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    /*=============================== TRIGGER SYSCALL ===========================*/
+    /*
+      PIDFD_NONBLOCK is available only on kernal versions > 5.10.00, hence used O_NONBLOCK 
+      See https://elixir.bootlin.com/linux/v5.10.185/source/include/uapi/linux/pidfd.h#L10
+    */
+    
+    int flags = 0;
+#if (LINUX_VERSION_CODE > KERNEL_VERSION(5, 10, 0))
+    flags = O_NONBLOCK;
+#endif
```
Comment (Andreagit97) [inline test/drivers/test_suites/syscall_exit_suite/pidfd_open_x.cpp:25]:
```suggestion
#ifdef PIDFD_NONBLOCK
    flags = PIDFD_NONBLOCK;
#else
    flags = O_NONBLOCK;
#endif 
```

**Review by Andreagit97 — COMMENTED** [review]
Thank you for this!

File: driver/bpf/fillers.h [hunk 2]
```diff
@@ -7051,4 +7051,27 @@ FILLER(sys_pidfd_getfd_x, true)
 	return bpf_push_u32_to_ring(data, val);
 }
 
+FILLER(sys_pidfd_open_x, true)
+{
+	int retval;
+	unsigned long val;
+	unsigned long res;
+	unsigned long flags;
```
[the remark this answers is not in the document]
Comment (Rohith-Raju) [reply driver/bpf/fillers.h:7059]: Sure, I've used the same declaration for `memfd_create` and `pidfd_open` as well. Do I change that as well?

File: driver/ppm_flag_helpers.h [hunk 3]
```diff
@@ -2078,6 +2081,25 @@ static __always_inline uint32_t splice_flags_to_scap(uint32_t flags)
...
+	PIDFD_NONBLOCK is available only on kernal versions > 5.10.00, hence used O_NONBLOCK 
+	See https://elixir.bootlin.com/linux/v5.10.185/source/include/uapi/linux/pidfd.h#L10
+*/
+    
+#ifdef O_NONBLOCK
+	if(flags & O_NONBLOCK) res |= PPM_PIDFD_NONBLOCK;
+#endif
+
+	return res;
+}
```
[the remark this answers is not in the document]
Comment (Rohith-Raju) [reply driver/ppm_flag_helpers.h:2101]: Sure!!

File: driver/bpf/fillers.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (Andreagit97) [reply driver/bpf/fillers.h:7059]: If you have time it would be great!

File: driver/bpf/fillers.h [hunk 4]
```diff
@@ -7051,4 +7051,22 @@ FILLER(sys_pidfd_getfd_x, true)
...
+{
+
+	/* Parameter 1: ret (type: PT_FD) */
+	long retval = bpf_syscall_get_retval(data->ctx);
+	int res = bpf_push_s64_to_ring(data, retval);
+	CHECK_RES(res);
+
+	/* Parameter 2: pid (type: PT_PID)*/
+	pid_t pid = (s32)bpf_syscall_get_argument(data, 0);
+	int res = bpf_push_s64_to_ring(data, (s64)pid);
```
Comment (Andreagit97) [inline driver/bpf/fillers.h:7064]:
```suggestion
	res = bpf_push_s64_to_ring(data, (s64)pid);
```

File: driver/ppm_fillers.c [hunk 5]
```diff
@@ -8142,3 +8143,29 @@ int f_sys_pidfd_getfd_x(struct event_filler_arguments *args){
 	
 	return add_sentinel(args);
 }
+
+int f_sys_pidfd_open_x(struct event_filler_arguments *args)
+{
+	unsigned long val;
+	int res;
+	int retval;
```
[the remark this answers is not in the document]
Comment (Andreagit97) [reply driver/ppm_fillers.c:8151]: we still need to address this

**Review by Andreagit97 — COMMENTED** [review]
Almost done :)

File: driver/ppm_fillers.c (hunk 5 above)
[the remark this answers is not in the document]
Comment (Rohith-Raju) [reply driver/ppm_fillers.c]: Oops forgot to commit 😅

**Comment by Rohith-Raju** [comment]
@Andreagit97 We probably need to add a kernel check since the test is using `5.8`. WDYT?

**Comment by Andreagit97** [comment]
> @Andreagit97 We probably need to add a kernel check since the test is using `5.8`. WDYT? https://app.circleci.com/pipelines/github/falcosecurity/libs/1551/workflows/e87d59c2-1033-4f49-9abd-de04f3d09db1/jobs/4382?invite=true#step-105-11153_6

I moved the flag checks in the failure case, this should solve our issues :crossed_fingers:

**Review by Andreagit97 — APPROVED** [review]
/approve

**Review by jasondellaluce — APPROVED** [review]
/approve

**Pull request merged**